tests/ui/cypress/specs/multi_user_openid_connect.js

```diff
@@ -0,0 +1,289 @@
+          cy.contains(
+            'Your user does not have sufficient rights to perform this action. Please contact your administrator.'
+          )
+          cy.contains('return to the login page')
+        }
+      }
+    )
+  })
+
+  it('Can logout', () => {
+    // login
+    cy.visit('/')
+    cy.url().should('not.include', Cypress.config('baseUrl'))
+    cy.contains('Mock OpenID Connect login form')
+    cy.get('input[name="username"]').clear().type(admin.u)
+    cy.contains('Sign In').click()
+
+    // verify that we are shown to be logged in to the Krill UI
+    cy.contains('Sign In').should('not.exist')
+    cy.url().should('include', Cypress.config('baseUrl'))
+    cy.get('#userinfo').click()
+    cy.get('#userinfo_table').contains(admin.u)
+
+    // logout
+    cy.intercept('GET', /^http:\/\/localhost:1818\/logout.+/).as('oidcLogout')
+    cy.get('.logout').click()
+    cy.wait('@oidcLogout').its('response.statusCode').should('eq', 302)
+
+    // verify that we are shown the OpenID Connect provider login page
+    cy.url().should('not.include', Cypress.config('baseUrl'))
+    cy.contains('Mock OpenID Connect login form')
+    cy.get('input[name="username"]')
+  })
+
+  it('Login with short-lived non-refreshable token and try to refresh page', () => {
+    cy.intercept('GET', '/api/v1/authorized').as('isAuthorized')
+    cy.visit('/')
+
+    cy.wait('@isAuthorized').its('response.statusCode').should('eq', 403)
+    cy.url().should('not.include', Cypress.config('baseUrl'))
+    cy.contains('Mock OpenID Connect login form')
+    cy.get('input[name="username"]').clear().type(shorttoken.u)
+
+    cy.intercept('GET', '/index.html').as('postLoginIndexFetch')
+    cy.intercept('GET', '/api/v1/authorized').as('isAuthorized')
+    cy.contains('Sign In').click()
+
+    cy.wait('@postLoginIndexFetch').its('response.statusCode').should('eq', 200)
+    cy.wait('@isAuthorized').its('response.statusCode').should('eq', 200)
+    cy.url().should('include', Cypress.config('baseUrl'))
+    cy.contains('Sign In').should('not.exist')
+    cy.get('#userinfo').click()
+    cy.get('#userinfo_table').contains(shorttoken.u)
+    cy.contains(shorttoken.u)
+    cy.contains('Welcome to Krill')
+
+    // the token has a lifetime of 5 second and no refresh token
+    // wait 6 seconds...
+    // note: a shorter token with a 1 second lifetime doesn't work in the GitHub
+    // Action runner environment because the token has sometimes already expired
+    // by the time Krill verifies it!
+    cy.wait(6000)
+
+    // Try to create a CA, by typing in the input, clicking the 'Create CA' button
+    // and then clicking 'Ok'. This should fail, since the token can't be refereshed.
+    cy.intercept('POST', '/api/v1/cas').as('createCA')
+    cy.get('main input').type('some-handle-name')
+    cy.get('main button').click()
+    cy.get('.el-message-box__btns button:nth-child(2)').click()
+    cy.wait('@createCA').its('response.statusCode').should('eq', 401)
```

Comment:
The appearance of the error message could be improved. That's not something to fix in this PR but I noticed it when seeing what the end user experience looks like with Cypress running the test interactively:
![image](https://user-images.githubusercontent.com/3304436/105863580-51427900-5ff1-11eb-985e-36bae6d27347.png)
At the very least perhaps we should make the text left-aligned so that it doesn't space out so much.

Reply:
I think Lagosta should be modified to handle the various errors more gracefully, like always supplying a suggested action to the user. Ideally it should not dump the API error it receives on the user directly (in the console is fine I guess)

Reply:
I think indeed a review of the current error handling and presentation strategy could yiled improvement, but I don't know if we have time for that before we release this, so I was thinking of a minor improvement for now.

Reply:
To be clear, I don't think this PR should be blocked by these issues, the primary value of this PR is handling OpenID Connect provider errors properly, and it would be unfair to say that this PR introduces the issues with the way errors are communicated to the Lagosta web UI and presented to the user.